src/battery-check/battery-check.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "battery-util.h"
+#include "build.h"
+#include "constants.h"
+#include "errno-util.h"
+#include "glyph-util.h"
+#include "fd-util.h"
+#include "io-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "socket-util.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "Checks battery level to see whether there's enough charge.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static void battery_check_send_plymouth_message(char *message) {
+        assert(message);
+
+        int r;
+        static const union sockaddr_union sa = PLYMOUTH_SOCKET;
+        _cleanup_close_ int fd = -EBADF;
+        _cleanup_free_ char *plymouth_message = NULL;
+        const char *new_mode = "shutdown";
+
+        int c = asprintf(&plymouth_message,
+                                 "M\x02%c%s%c"
+                                 "C\x02%c%s",
+                                 (int) strlen(message) + 1, message, '\x00', (int) strlen(new_mode),
+new_mode);
```

Comment:
Ah, looking again, I think I see an (the?) issue. Changing modes clears messages from the previous mode, so the order needs to be "switch to shutdown mode, show message" and this code is doing "show message, switch to shutdown mode"

I think this may help?

```suggestion
        int c = asprintf(&plymouth_message,
                                 "C\x02%c%s%c"
                                 "M\x02%c%s",
                                 (int) strlen(new_mode) + 1, new_mode, '\x00',
                                 (int) strlen(message), message);
```